src/main/java/com/rapid7/net/LoggerConfiguration.java

```diff
@@ -0,0 +1,214 @@
+public class LoggerConfiguration {
+
+    /**
+     * Destination Token.
+     */
+    private String token;
+    /**
+     * Destination Region.
+     */
+    private String region;
+    /**
+     * Account Key.
+     */
+    private String key;
+    /**
+     * Account Log Location.
+     */
+    private String location;
+    /**
+     * HttpPut flag.
+     */
+    private boolean httpPut;
+    /**
+     * SSL/TLS flag.
+     */
+    private boolean ssl;
+    /**
+     * Debug flag.
+     */
+    private boolean debug;
+    /**
+     * UseDataHub flag.
+     */
+    private boolean useDataHub;
+    /**
+     * DataHubAddr - address of the server where DataHub instance resides.
+     */
+    private String dataHubAddr;
+    /**
+     * DataHubPort - port on which DataHub instance waits for messages.
+     */
+    private int dataHubPort;
+    /**
+     * LogHostName - switch that determines whether HostName should be appended to the log message.
+     */
+    private boolean logHostName;
+    /**
+     * HostName - value, that should be appended to the log message if logHostName is set to true.
+     */
+    private String hostName;
+    /**
+     * LogID - user-defined ID string that is appended to the log message if non-empty.
+     */
+    private String logID;
+
+    private LoggerConfiguration() {
+    }
+
+    public String getToken() {
+        return token;
+    }
+
+    public String getKey() {
+        return key;
+    }
+
+    public String getLocation() {
+        return location;
+    }
+
+    public boolean isHttpPut() {
+        return httpPut;
+    }
+
+    public boolean isSsl() {
+        return ssl;
+    }
+
+    public boolean isDebug() {
+        return debug;
+    }
+
+    public boolean isUseDataHub() {
+        return useDataHub;
+    }
+
+    public String getDataHubAddr() {
+        return dataHubAddr;
+    }
+
+    public int getDataHubPort() {
+        return dataHubPort;
+    }
+
+    public boolean isLogHostName() {
+        return logHostName;
+    }
+
+    public String getHostName() {
+        return hostName;
+    }
+
+    public String getLogID() {
+        return logID;
+    }
+
+    public String getRegion() {
+        return region;
+    }
+
+    public static class Builder {
+        private String token;
+        private String region;
+        private String key;
+        private String location = "";
+        private boolean httpPut;
+        private boolean ssl = true;
+        private boolean debug;
+        private boolean useDataHub;
+        private String dataHubAddr;
+        private int dataHubPort;
+        private boolean logHostName;
+        private String hostName;
+        private String logID;
+
+        public Builder useToken(String token) {
+            this.token = token;
+            return this;
+        }
+
+        public Builder inRegion(String region) {
+            this.region = region;
+            return this;
+        }
+
+        public Builder withHttpPutKey(String key) {
```

Comment:
Any reason this is the only method using `with` naming?

Reply:
no reason, don't you like it?

Reply:
For consistency, let's leave it out. Better to use it nowhere or everywhere.